core/src/main/java/org/apache/spark/io/ReadAheadInputStream.java

```diff
@@ -78,9 +79,8 @@
   // whether there is a read ahead task running,
   private boolean isReading;
 
-  // If the remaining data size in the current buffer is below this threshold,
-  // we issue an async read from the underlying input stream.
-  private final int readAheadThresholdInBytes;
+  // whether there is a reader waiting for data.
+  private AtomicBoolean isWaiting = new AtomicBoolean(false);
```

Comment:
You can just use `volatile` here

Reply:
I'll leave it be - should compile to basically the same, and with using `AtomicBoolean` the intent seems more readable to me.